src/detectors/builtin/sendInLoop.ts

```diff
@@ -0,0 +1,184 @@
+import { CompilationUnit } from "../../internals/ir";
+import { foldStatements, isSelf } from "../../internals/tact";
+import { MistiTactWarning, Severity } from "../../internals/warnings";
+import { ASTDetector } from "../detector";
+import {
+  AstStatement,
+  AstExpression,
+  AstCondition,
+  AstStatementTry,
+  AstStatementTryCatch,
+  AstStatementWhile,
+  AstStatementRepeat,
+  AstStatementUntil,
+  AstStatementForEach,
+  idText,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+export class SendInLoop extends ASTDetector {
+  severity = Severity.HIGH;
+
+  async check(cu: CompilationUnit): Promise<MistiTactWarning[]> {
+    const warnings: MistiTactWarning[] = [];
+    this.collectWarnings(cu, warnings);
+    return warnings;
+  }
+
+  private collectWarnings(
+    cu: CompilationUnit,
+    warnings: MistiTactWarning[],
+  ): void {
+    const processedLoopIds = new Set<number>();
+    for (const node of cu.ast.getProgramEntries()) {
```

Comment:
```suggestion
Array.from(cu.ast.getProgramEntries()).reduce( ... 
```

And it should be moved to the `checkFunction`